File: UVA/Division.py
import sys, math
sys.stdin = open("in.txt", "r")
sys.stdout = open("out.txt", "w")

while True:
	try:
		nums = []
		S = input().split()
		for i in S: nums.append(int(i))
		x = (nums[0]**nums[1]-1)
		y = (nums[0]**nums[2]-1)
		res = x//y
		print("(",nums[0],"^"+str(nums[1])+"-1)/("+str(nums[0])+"^"+str(nums[2])+"-1) ", end="", sep="")
		if(x%y==0 and math.log10(res)<=100): print(int(res))
		else: print("is not an integer with less than 100 digits.")
	except:
		break;

File: UVA/Division.java
import java.io.NotActiveException;
import java.math.BigInteger;
import java.util.Scanner;

import javax.swing.event.AncestorEvent;
import javax.swing.text.html.HTMLDocument.HTMLReader.IsindexAction;

class Product
{
	public static void main(String args[]) {
		int t, a, b;
		BigInteger x, y;
		String res;
		Scanner cin = new Scanner(System.in);

		while(cin.hasNext()){
			t = cin.nextInt();
			a = cin.nextInt();
			b = cin.nextInt();
			System.out.print("(" + t + "^"+ a + "-1)"+ "/" + "(" + t + "^" + b + "-1) ");
			if(t==1){
				System.out.println("is not an integer with less than 100 digits.");
				continue;
			}

			if(a==b){
				System.out.println("1");
			}

			x = BigInteger.valueOf(t).pow(a);
			y = BigInteger.valueOf(t).pow(b);
			x = x.add(BigInteger.valueOf(-1));
			y = y.add(BigInteger.valueOf(-1));
			BigInteger m = x.mod(y);
			if(m.equals(0)) {
				x = x.divide(y);
				res = x.toString();
				if(res.length()<100) System.out.println(x);
				else System.out.println("is not an integer with less than 100 digits.");
				continue;
			}
			System.out.println("is not an integer with less than 100 digits.");
		}
	}
}
